packages/SwingSet/src/kernel/kernel.js

```diff
@@ -634,6 +634,58 @@ export default function buildKernel(
     );
   }
 
+  function setKernelVatOption(vatID, option, value) {
+    switch (option) {
+      case 'reapInterval': {
+        if (
+          typeof value === 'number' ||
+          typeof value === 'bigint' ||
+          value === 'never'
+        ) {
+          const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
+          vatKeeper.updateReapInterval(value);
+        } else {
+          console.log(`WARNING: invalid reapInterval value`, value);
+        }
+        return true;
+      }
+      default:
+        return false;
+    }
+  }
+
+  /**
+   *
+   * @param { RunQueueEventChangeVatOptions } message
+   * @returns { Promise<DeliveryStatus | null> }
+   */
+  async function processChangeVatOptions(message) {
+    const { vatID, options } = message;
+    insistVatID(vatID);
+    // eslint-disable-next-line no-use-before-define
+    assert(vatWarehouse.lookup(vatID));
```

Comment:
oh, does this tolerate a vat being terminated before the `changeVatOptions` event is processed?

Reply:
No.  I guess these non-send deliveries don't benefit from the automatic message splatting that regular message deliveries get.

I cribbed a lot from `processUpgradeVat` and I notice that it has the same flaw.  I will fix them both.

Also: `processStartVat` kinda sorta has the same issue, but I don't think a vat can be terminated before it is started -- though as I wrote that I started to wonder...